<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PciRsiPlannerDialogBase</class>
 <widget class="QDialog" name="PciRsiPlannerDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>750</width>
    <height>576</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>576</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PCI / RSI Planner</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icon_planner.svg</normaloff>icon_planner.svg</iconset>
  </property>
  <layout class="QVBoxLayout" name="mainVerticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="leftLayout">
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="layerLabel">
           <property name="text">
            <string>Layer:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="layerComboBox"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="techLabel">
           <property name="text">
            <string>Technology field (LTE/NR):</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="techFieldComboBox"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="siteIdLabel">
           <property name="text">
            <string>eNB/gNB ID field:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="siteIdFieldComboBox"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="cellIdLabel">
           <property name="text">
            <string>Cell/Sector ID field:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="cellIdFieldComboBox"/>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="bandLabel">
           <property name="text">
            <string>Band field:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="bandFieldComboBox"/>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="earfcnLabel">
           <property name="text">
            <string>EARFCN / NR-ARFCN field:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QComboBox" name="earfcnFieldComboBox"/>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="lockedLabel">
           <property name="text">
            <string>Locked field (optional):</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QComboBox" name="lockedFieldComboBox">
           <item>
            <property name="text">
             <string>(None)</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="existingPciLabel">
           <property name="text">
            <string>Existing PCI field (optional):</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QComboBox" name="existingPciFieldComboBox"/>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="existingRsiLabel">
           <property name="text">
            <string>Existing RSI field (optional):</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QComboBox" name="existingRsiFieldComboBox"/>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="pciPlanFieldLabel">
           <property name="text">
            <string>New PCI plan field name:</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QLineEdit" name="pciPlanFieldLineEdit">
           <property name="text">
            <string>PCI_PLAN</string>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QLabel" name="rsiPlanFieldLabel">
           <property name="text">
            <string>New RSI plan field name:</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QLineEdit" name="rsiPlanFieldLineEdit">
           <property name="text">
            <string>RSI_PLAN</string>
           </property>
          </widget>
         </item>
         <item row="11" column="0">
          <widget class="QLabel" name="pciRangeLabel">
           <property name="text">
            <string>PCI range (min - max):</string>
           </property>
          </widget>
         </item>
         <item row="11" column="1">
          <layout class="QHBoxLayout" name="pciRangeLayout">
           <item>
            <widget class="QSpinBox" name="pciMinSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>503</number>
             </property>
             <property name="value">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="pciMaxSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>503</number>
             </property>
             <property name="value">
              <number>503</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="12" column="0">
          <widget class="QLabel" name="rsiRangeLabel">
           <property name="text">
            <string>RSI range (min - max):</string>
           </property>
          </widget>
         </item>
         <item row="12" column="1">
          <layout class="QHBoxLayout" name="rsiRangeLayout">
           <item>
            <widget class="QSpinBox" name="rsiMinSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>1007</number>
             </property>
             <property name="value">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="rsiMaxSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>1007</number>
             </property>
             <property name="value">
              <number>1007</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="13" column="0">
          <widget class="QLabel" name="reuseDistanceLabel">
           <property name="text">
            <string>Minimum Reuse Distance (km):</string>
           </property>
          </widget>
         </item>
         <item row="13" column="1">
          <widget class="QDoubleSpinBox" name="reuseDistanceSpinBox">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>0.500000000000000</double>
           </property>
           <property name="maximum">
            <double>50.000000000000000</double>
           </property>
           <property name="value">
            <double>10.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="14" column="0">
          <widget class="QLabel" name="planningOptionsLabel">
           <property name="text">
            <string>Planning Options:</string>
           </property>
          </widget>
         </item>
         <item row="14" column="1">
          <layout class="QHBoxLayout" name="planningOptionsLayout">
           <item>
            <widget class="QCheckBox" name="planPciCheckBox">
             <property name="text">
              <string>Plan PCI</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="planRsiCheckBox">
             <property name="text">
              <string>Plan RSI</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="15" column="0">
          <widget class="QLabel" name="pciModCheckLabel">
           <property name="text">
            <string>PCI Mod 3/6 Conflict Check:</string>
           </property>
          </widget>
         </item>
         <item row="15" column="1">
          <widget class="QCheckBox" name="pciModCheckBox">
           <property name="text">
            <string>Enable (recommended)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QTextBrowser" name="infoTextBrowser">
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;h3 style=&quot; margin-top:10px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt; font-weight:600; color:#2196f3;&quot;&gt;PCI/RSI Planner&lt;/span&gt;&lt;/h3&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Purpose:&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; Automatically assigns Physical Cell IDs (PCI) and Root Sequence Indexes (RSI) per technology and band.&lt;/span&gt; &lt;/p&gt;
&lt;h3 style=&quot; margin-top:10px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt; font-weight:600; color:#2196f3;&quot;&gt;Sample Data:&lt;/span&gt;&lt;/h3&gt;
&lt;p style=&quot; margin-top:3px; margin-bottom:3px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:7.5pt;&quot;&gt;TECH | ENB_ID | CELL_ID | BAND&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:3px; margin-bottom:3px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:7.5pt;&quot;&gt;LTE | 12345 | 0 | L1800&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:3px; margin-bottom:3px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:7.5pt;&quot;&gt;NR | 67890 | 0 | n78&lt;/span&gt;&lt;/p&gt;
&lt;h3 style=&quot; margin-top:10px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt; font-weight:600; color:#2196f3;&quot;&gt;Instructions:&lt;/span&gt;&lt;/h3&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;1. Select layer with sectors&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;2. Map technology and band fields&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;3. Optionally map existing PCI/RSI and locked fields (locked: 1=locked, 0=unlocked)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;4. Set PCI range (e.g., 0-503 for LTE)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;5. Set RSI range (e.g., 0-837 for LTE)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;6. Click Run Planner&lt;/span&gt; &lt;/p&gt;
&lt;h3 style=&quot; margin-top:10px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt; font-weight:600; color:#2196f3;&quot;&gt;Features:&lt;/span&gt;&lt;/h3&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot;&quot; style=&quot; margin-top:12px; margin-bottom:0px; margin-left:15px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Per-band planning&lt;/li&gt;
&lt;li style=&quot;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:15px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Respects locked cells&lt;/li&gt;
&lt;li style=&quot;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:15px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Reuse distance checking&lt;/li&gt;
&lt;li style=&quot;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:15px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;PCI mod 3/6 conflict avoidance&lt;/li&gt;
&lt;li style=&quot;&quot; style=&quot; margin-top:0px; margin-bottom:12px; margin-left:15px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Geographic distance aware &lt;/li&gt;&lt;/ul&gt;
&lt;h3 style=&quot; margin-top:10px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt; font-weight:600; color:#2196f3;&quot;&gt;Tips:&lt;/span&gt;&lt;/h3&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;• LTE: PCI 0-503, RSI 0-837&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;• 5G NR: PCI 0-1007&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;• Locked field: 1, True, or any non-zero value = locked; 0, False, or empty = unlocked&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:6px; margin-bottom:6px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;• Locked cells keep their existing PCI/RSI values&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="progressLayout">
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="visible">
        <bool>true</bool>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="textVisible">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="bottomLayout">
     <property name="spacing">
      <number>6</number>
     </property>
     <item>
      <widget class="QPushButton" name="runButton">
       <property name="text">
        <string>Run Planner</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Close</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PciRsiPlannerDialogBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
